### ISSUE TITLE
networkmanager pulls in gtk3+ by default, can't seem to disable it either

### ISSUE THREAD

**Issue by colemickens** [NixOS/nixpkgs#196067]
### Describe the bug

see: #164531

I have this in my nixos config:

```
networking.networkmanager.plugins = lib.mkForce [];
```

yet I can see NM pulling in OpenConnect and GTK3.

I think we should reconsider if NM Should pull in GTK3 by default, and make it easier (possible?) to actually disable these plugins. (Apologies if I'm just missing something obvious here)

cc: @jtojnar

### Steps To Reproduce
1. Try to use NM with GTK3

### Expected behavior
It's possible.

**Comment by jtojnar** [comment]
NetworkManager contains some VPN-specific code in the project itself. I talked with the developers about moving it into the plug-ins few years back and they said they would be fine with it but I have not had time to work on it yet (and probably will not in foreseeable future).

**Comment by colemickens** [comment]
Oh, are you saying that NM requires OpenConnect even without actually trying to opt-in to it? I guess that makes sense as to how I'm seeing this post #164531... I'll look around more with that in mind. Thanks!

**Comment by jtojnar** [comment]
Yeah https://github.com/NixOS/nixpkgs/blob/0ece45b46807ff01ab6dab634fcc7b49d6c73ac1/pkgs/tools/networking/networkmanager/default.nix#L118-L123

**Comment by Artturin** [comment]
adding a `withGtk3` arg would make it easier to disable the dep

**Comment by BeneSim** [comment]
> Here is a simple workaround for this issue that I've found. In addition to :
> 
> networking.networkmanager.plugins = lib.mkForce [];
> add:
> 
> nixpkgs.overlays = [(final: prev: { openconnect = null; })];
> It simply removes the `openconnect` package using the overlay mechanism. `networkmanager` ofc needs to be rebuilt. But the resulting package doesn't depend any more on `gtk+`.
> 
> I use it with 24.11.

Seems to have been fixed on master https://github.com/NixOS/nixpkgs/commit/c6eb84310af667b270b4a07095ce955639fbfaf2 . 

Until then you could also directly override the openconnect parameter for the networkmanager package instead of overlaying openconnect globally, e.g.
```nix
nixpkgs.overlays = [(final: prev: { networkmanager = prev.networkmanager.overrde { openconnect = null; }; })];
```
or even go as far and only override the networkmanager package that the networkmanager service will use (i.e. `networking.networkmanager.package = pkgs.networkmanager.override { ... };`)
